includes/User.php

```diff
@@ -2367,7 +2367,13 @@ public function getOption($oname, $defaultOverride = null, $ignoreHidden = false
 	 */
 	private function getOptionHelper( $oname, $defaultOverride = null, $ignoreHidden = false ) {
 		global $wgHiddenPrefs;
+		global $wgPrivateUserAttributes;
+
 		$this->loadOptions();
+		if ( in_array( $oname, $wgPrivateUserAttributes ) && array_key_exists( $oname, $this->mOptions ) ) {
+			$options = $this->getOptions();
+			return array_key_exists( $oname, $options ) ? $options[$oname] : $defaultOverride;
+		}
```

Comment:
I don't understand why this change is needed. If we save a property to the db and `user-preferences`, just reading from the db should be fine. Could you explain this change?

Reply:
This one fixes saving options to user-preferences from Spcial:Preferences which were modified on ucp

Reply:
Probably it is caching problem